tcms/xmlrpc/tests/test_testrun.py

```diff
@@ -202,3 +203,64 @@ def test_create_with_product_version(self):
         # the result is still using product_version from TR.plan.product_version
         # not the one we specified above
         self.assertEqual(result['product_version'], self.plan.product_version.value)
+
+
+class TestUpdateTestRun(XmlrpcAPIBaseTest):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.test_run = TestRunFactory()
+
+        self.updated_test_plan = TestPlanFactory()
+        self.updated_build = BuildFactory()
+        self.updated_summary = 'Updated summary.'
+        self.updated_stop_date = '2020-05-05'
+
+    def test_successful_update(self):
+        update_fields = {
+            'plan': self.updated_test_plan.pk,
+            'build': self.updated_build.pk,
+            'summary': self.updated_summary,
+            'stop_date': self.updated_stop_date
+        }
+
+        # assert test plan is not already updated
+        self.assertNotEqual(self.updated_test_plan, self.test_run.plan.name)
+        self.assertNotEqual(self.updated_build, self.test_run.build.name)
+        self.assertNotEqual(self.updated_summary, self.test_run.summary)
+        self.assertNotEqual(self.updated_stop_date, str(self.test_run.stop_date))
+
+        updated_test_run = self.rpc_client.exec.TestRun.update(self.test_run.pk, update_fields)
+        self.test_run.refresh_from_db()
+
+        # compare result, returned form API call with test run from DB
+        self.assertEqual(updated_test_run['plan'], self.test_run.plan.name)
+        self.assertEqual(updated_test_run['build'], self.test_run.build.name)
+        self.assertEqual(updated_test_run['summary'], self.test_run.summary)
+        self.assertEqual(updated_test_run['stop_date'], str(self.test_run.stop_date))
+
+        # compare result, returned form API call with params sent to the API
+        self.assertEqual(updated_test_run['plan'], self.updated_test_plan.name)
+        self.assertEqual(updated_test_run['build'], self.updated_build.name)
+        self.assertEqual(updated_test_run['summary'], self.updated_summary)
+        self.assertEqual(
+            updated_test_run['stop_date'],
+            str(datetime.strptime(self.updated_stop_date, '%Y-%m-%d'))
```

Comment:
Doesn't either `str` or `strptime` alone work? See the above group of asserts.

Reply:
it did, but I had to tweek the data a bit. 
now `self.updated_stop_date = datetime.strptime('2020-05-05', '%Y-%m-%d')`
and before it was just `self.updated_stop_date = '2020-05-05'`